packages/core/content-manager/server/services/entity-manager.js

```diff
@@ -385,4 +385,24 @@ module.exports = ({ strapi }) => ({
 
     return sumDraftCounts(entity, uid);
   },
+
+  async getMultipleEntriesNumberOfDraftRelations(ids, uid) {
+    const { populate, hasRelations } = getDeepPopulateDraftCount(uid);
+
+    if (!hasRelations) {
+      return 0;
+    }
+
+    const entities = await strapi.entityService.findMany(uid, {
+      populate,
+    });
+
+    const totalNumberDraftRelations = entities.reduce(async (acc, entity) => {
+      const sumDraftRelations = sumDraftCounts(entity, uid);
+      const prevCounter = await acc;
+      return sumDraftRelations + prevCounter;
+    }, 0);
```

Comment:
we have a utility to work with reduce async that it m ight be of help here :) 

```js
const { reduceAsync } = require('@strapi/utils');

const totalNumberDraftRelations = await reduceAsync(entities)((count, entity) => {
      return sumDraftCounts(entity, uid) + count;
    }, 0);
```

In any case, does this need to be async? I think sumDraftCounts does not return a promise.

```suggestion
    return entities.reduce((count, entity) => sumDraftCounts(entity, uid) + count, 0);
```

Reply:
WOW, it solves all my problems and also it is really smooth, thanks a lot @Marc-Roig